yellow-paper/docs/circuits/private-kernel-initial.md

```diff
@@ -0,0 +1,258 @@
+:::
+
+## Requirements
+
+In the **initial** kernel iteration, the process involves taking a transaction request and private call data, verifying their integrity, and preparing the necessary data for subsequent circuits to operate. This step is particularly beneficial due to its separation from the [inner private kernel circuit](./private-kernel-inner.md), as the first call lacks a "previous kernel" to process. Additionally, it executes tasks that are pertinent to a transaction and need only occur once.
+
+### Key Responsibilities Specific to this Circuit
+
+#### Validating the correspondence of function call with caller's intent.
+
+This entails ensuring that the following data from the private call aligns with the specifications in the transaction request:
+
+- Contract address.
+- [Function data](#function_data).
+- Function arguments.
+
+> Although it's not enforced in the protocol, it is customary to provide a signature signed over the transaction request and verify it in the first function call. This practice guarantees that only the party possessing the key(s) can authorize a transaction with the exact transaction request.
+
+#### Verifying the legitimacy of the function as the entrypoint.
+
+- It must be a private function.
+- It must not be an internal function.
+
+#### Ensuring the function call is the first call.
+
+- It must not be a delegate call.
+- It must not be a static call.
+
+#### Ensuring transaction uniqueness.
+
+- It must emit the hash of the transaction request as the **first** nullifier.
+
+This nullifier serves multiple purposes:
+
+- Identifying a transaction.
+- Preventing the signature of a transaction request from being reused in another transaction.
+- Generating values that should be maintained within the transaction's scope. For example, it is utilized to compute the nonces for all the note hashes in a transaction.
+
+> Note that the final transaction data is not deterministic for a given transaction request. The production of new notes, the destruction of notes, and various other values are likely to change based on the time and conditions when a transaction is being composed. However, the intricacies of implementation should not be a concern for the entity initiating the transaction.
+
+### Processing Private Function Call
+
+#### Ensuring the contract instance being called is deployed.
+
+It proves that the nullifier representing the contract exists in the contract tree.
+
+This nullifier is the contract address siloed with the address of a precompiled deployment contract.
+
+#### Ensuring the function being called exists in the contract.
+
+The contract address contains the contract class ID, which is a hash of the root of its function tree and additional values. This circuit leverages these characteristics to establish the validity of the function's association with the contract address.
+
+Each leaf of the function tree is a hash representing a function. The preimage includes:
```

Comment:
"includes" is a bit vague - does the preimage contain other information not listed here? Ideally the list is complete and if it is, we can replace "includes:" with "is defined as the concatenation of:"